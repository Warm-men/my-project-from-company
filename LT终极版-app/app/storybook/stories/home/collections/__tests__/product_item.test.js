import React from 'react'
import render from 'react-test-renderer'
import ProductItem from '../product_item'
import { shallow } from 'enzyme'

import AddToClosetButton, * as add_to_closet_button from '../../../../../src/containers/closet/add_to_closet_button'
import ToteSlot from '../../../products/tote_slot'

add_to_closet_button.default = jest.fn().mockImplementation(() => {
  return {
    render: () => <div>MockComponent</div>
  }
})

describe('test product item in new arrival collection', () => {
  let props
  let wrapper
  let get_report_data
  let did_selected_item
  beforeEach(() => {
    get_report_data = jest.fn()
    did_selected_item = jest.fn()
    props = {
      data: {
        title: 'tt',
        catalogue_photos: [
          {
            full_url: 'xxx'
          }
        ],
        tote_slot: 1,
        type: 'Clothing'
      },
      column: 'xx'
    }

    wrapper = shallow(
      <ProductItem
        data={props.data}
        column={props.column}
        getReportData={get_report_data}
        didSelectedItem={did_selected_item}
      />
    )
  })
  it('should render with right style', () => {
    const tree = render
      .create(
        <ProductItem
          data={props.data}
          column={props.column}
          getReportData={jest.fn()}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('test tote slot', () => {
    it('should not render ToteSlot when tote slot is 1', () => {
      const tote_slot = wrapper.find(ToteSlot)
      expect(tote_slot).toHaveLength(0)
    })

    it('should render ToteSlot when tote slot is 2', () => {
      wrapper.setProps({
        data: {
          title: 'tt',
          catalogue_photos: [
            {
              full_url: 'xxx'
            }
          ],
          tote_slot: 2,
          type: 'Clothing'
        }
      })
      const tote_slot = wrapper.find(ToteSlot)
      expect(tote_slot.props().slotNum).toEqual(2)
    })
  })

  describe('test add to closet button', () => {
    it('should set right buttonStyle', () => {
      const add_to_closet_btn = wrapper.find(AddToClosetButton)
      expect(add_to_closet_btn.props().product).toEqual(props.data)
    })

    it('type是否有被传入', () => {
      wrapper.setProps({
        data: {
          title: 'tt',
          catalogue_photos: [
            {
              full_url: 'xxx'
            }
          ],
          tote_slot: 2,
          type: 'Clothing'
        }
      })
      expect(wrapper.find({ testID: 'tote-slot' }).props().type).toEqual(
        'Clothing'
      )
    })

    it('should call report data when click', () => {
      const add_to_closet_btn = wrapper.find(AddToClosetButton)
      add_to_closet_btn.props().getReportData()
      expect(get_report_data.mock.calls).toHaveLength(1)
    })

    it('should call report data when click', () => {
      const add_to_closet_btn = wrapper.find(AddToClosetButton)
      expect(add_to_closet_btn.props().buttonStyle).toEqual({
        position: 'absolute',
        top: 0,
        right: 0,
        padding: 8,
        transform: [{ scale: 0.8 }]
      })
    })
  })

  it('should call did select when click', () => {
    wrapper
      .find({ testID: 'test-product-item' })
      .props()
      .onPress()
    expect(did_selected_item.mock.calls).toHaveLength(1)
    expect(did_selected_item.mock.calls[0][0]).toEqual(props.data)
  })
})
