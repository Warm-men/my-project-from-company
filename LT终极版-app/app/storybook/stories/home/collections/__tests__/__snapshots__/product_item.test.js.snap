// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test product item in new arrival collection should render with right style 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginBottom": 24,
      "marginRight": 16,
      "opacity": 1,
    }
  }
  testID="test-product-item"
>
  <View
    style={
      Array [
        Object {
          "height": 342,
          "marginTop": 1,
          "width": 228,
        },
        Object {
          "overflow": "hidden",
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      description="xx-tt"
      resizeMode="cover"
      source={
        Object {
          "uri": "xxx",
        }
      }
      style={
        Object {
          "height": 342,
          "width": 228,
        }
      }
    />
    <Text
      style={
        Object {
          "display": "none",
        }
      }
    >
      xx-tt
    </Text>
  </View>
  <div>
    MockComponent
  </div>
</View>
`;
